core/src/main/java/org/keycloak/Feature.java

```diff
@@ -0,0 +1,245 @@
+     *     - pom.xml (keycloak-parent - properties in 'community' and 'product' profiles)
+     *     - wildfly/server-subsystem/src/main/config/default-server-subsys-config.properties (<features> section)
+     *     - wildfly/server-subsystem/src/main/resources/subsystem-templates/keycloak-server.xml (<features> section)
+     *     - wildfly/server-subsystem/src/main/resources/cli/default-keycloak-subsys-config.cli
+     *     - wildfly/server-subsystem/src/test/java/org/keycloak/subsystem/server/extension/FeaturesTestCase.java (COMMUNITY_FEATURES, and PRODUCT_FEATURES constants)
+     *     - distribution/feature-packs/server-feature-pack/src/main/resources/content/bin/migrate-standalone.cli
+     *     - distribution/feature-packs/server-feature-pack/src/main/resources/content/bin/migrate-standalone-ha.cli
+     *     - distribution/feature-packs/server-feature-pack/src/main/resources/content/bin/migrate-domain-standalone.cli
+     *     - distribution/feature-packs/server-feature-pack/src/main/resources/content/bin/migrate-domain-clustered.cli
+     *     - distribution/distribution-tests/src/test/java/org/keycloak/test/distribution/DistributionFeaturesTest.java  (COMMUNITY_FEATURES, and PRODUCT_FEATURES constants)
+     *
+     *   In addition, the following files may need to be modified to support tests:
+     *     - testsuite/integration-arquillian/tests/base/src/test/resources/META-INF/keycloak-server.json (feature section)
+     *     - testsuite/utils/src/main/resources/META-INF/keycloak-server.json (feature section)
+     *
+     */
+
+    AUTHORIZATION(true), IMPERSONATION, SCRIPTS(true), DOCKER, ACCOUNT2(true), TOKEN_EXCHANGE;
+
+
+    private static final Logger log = Logger.getLogger(Feature.class.getName());
+
+    private boolean preview;
+
+    private static boolean foundBadConfig = false;
+
+    Feature() {
+        this.preview = false;
+    }
+
+    Feature(boolean preview) {
+        this.preview = preview;
+    }
+
+    public String caption() {
+        String s = name().toLowerCase().replace('_', '-');
+        return preview ? s + "-preview" : s;
+    }
+
+    public static Set<Feature> getDisabledFeatures() {
+        // by default all features are disabled
+        HashSet<Feature> disabled = new HashSet(Arrays.asList(values()));
+        Config.Scope config = Config.scope("feature");
+        for (Feature f: values()) {
+            Config.Scope featureScope = config.scope(f.caption());
+            if (!featureScope.getBoolean("enabled", false)) {
+                disabled.add(f);
+            } else {
+                disabled.remove(f);
+            }
+        }
+
+        return disabled;
+    }
+
+    public static boolean isFeatureEnabled(Feature feature) {
+        return Config.scope("feature", feature.caption()).getBoolean("enabled", false);
+    }
+
+    public static Feature fromCaption(String name) {
+        Feature ret = AUTHORIZATION.caption().equals(name) ? AUTHORIZATION :
```

Comment:
Why not just use Feature.values and loop through them? That way this doesn't have to be updated manually

Reply:
Doh! Of course :)